_source/_posts/2021-10-16-full-stack-java.md

```diff
@@ -0,0 +1,693 @@
+
+```
+jhipster heroku
+```
+
+This will start the [Heroku sub-generator](https://www.jhipster.tech/heroku/) that asks you a couple questions about your app: what you want to name it and whether you want to deploy it to a US region or EU. Then it'll prompt you to choose between building locally or with Git on Heroku's servers. Choose Git, so you don't have to upload a fat JAR. When prompted to use Okta for OIDC, type `N`. Then, the deployment process will begin.
+
+If you have a stable and fast internet connection, your app should be live on the internet in around six minutes!
+
+```
+remote: -----> Compressing...
+remote:        Done: 119.4M
+remote: -----> Launching...
+remote:        Released v6
+remote:        https://flickr-2.herokuapp.com/ deployed to Heroku
+remote:
+remote: Verifying deploy... done.
+
+To https://git.heroku.com/flickr-2.git
+
+ * [new branch]      HEAD -> main
+
+Your app should now be live. To view it run
+	heroku open
+And you can view the logs with this command
+	heroku logs --tail
+After application modification, redeploy it with
+	jhipster heroku
+Congratulations, JHipster execution is complete!
+Sponsored with ❤️  by @oktadev.
+Execution time: 6 min. 0 s.
+```
+
+### Configure for Auth0 and Analyze Your PWA Score with Lighthouse
+
+To configure your app to work with Auth0 on Heroku, run the following command to set your Auth0 variables on Heroku.
+
+```bash
+heroku config:set \
+  SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_OIDC_ISSUER_URI="https://<your-auth0-domain>/" \
+  SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_ID="<your-client-id>" \
+  SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET="<your-client-secret>" \
+  JHIPSTER_SECURITY_OAUTH2_AUDIENCE="https://<your-auth0-domain>/api/v2/"
+```
+
+Then, log in to your Auth0 account, navigate to your app, and add your Heroku URLs as valid redirect URIs:
+
+- Allowed Callback URLs: `https://flickr-2.herokuapp.com/login/oauth2/code/oidc`
+- Allowed Logout URLs: `https://flickr-2.herokuapp.com`
+
+After Heroku restarts your app, open it, and log in.
+
+{% img blog/full-stack-java/app-on-heroku.png alt:"Running on Heroku!" width:"800" %}{: .center-image }
+
+Then, test it with [Lighthouse](https://developers.google.com/web/tools/lighthouse/) (using the Lighthouse tab in Chrome developer tools). Looks pretty good, eh?! 💯
+
+{% img blog/full-stack-java/lighthouse-score.png alt:"Lighthouse Score 💯" width:"800" %}{: .center-image }
+
+## Learn More About Full Stack Java Development
+
+This tutorial shows you how to streamline full-stack Java development with JHipster. You developed a working application with a React frontend and a Spring Boot backend. You can find the app created in this tutorial on GitHub, in the [auth0-full-stack-java-example](https://github.com/oktadev/auth0-full-stack-java-example) repository.
+
+Since I didn't show you a lot about coding in Java, React, or Spring Boot, here are some helpful resources for that.
```

Comment:
I'm thinking whether this section with helpful resources should be in prerequisites or not. In that way, someone can go through those tutorials before starting analyzing this post. What do you think about it?

Reply:
In the past, we've tried not to give too many links in the beginning for fear of folks leaving the post and not coming back. I think this is a good idea though, so I'll make this change.